### DISCUSSION TITLE
You are awesome - keep going!

### DISCUSSION THREAD

**Discussion by mstahly** [Fruktus/QuadradiusPreservationProject#26] [General]
Dear Fruktus,

I've been checking in on this project for many months now and am inspired by your undertaking it. I love this game and played a ton with friends during the pandemic. Several of them have been monitoring your Reddit post as well. We are all very excited at the possibility of being able to play again. I just signed up for github so I could write this little message of encouragement. Also, a couple of us have basic coding skills and are happy to help with this project if ther'e anything you can outsource. Cheers, Mike

**Comment by Fruktus** [comment]
Hi, thank you so much for writing and glad you joined GitHub specifically to say this :D. This project is just a server for the OG version (not the reimplementation), and right now it works but is unable to download the OG files as they have been taken down. There is only a few cosmetic features missing, while gameplay should work in 100%. There is a second project for remaking the game, but due to copyright issues and lack of time have been gathering dust for some time, but I still hope to pick it up sometime. If you'd like to help, the best you could do is to set up this server with OG files and test it for bugs :) Best regards. Fruktus

**Reply by mstahly** [reply]
Thanks for responding and explaining. I'm happy to test for bugs. I haven't looked through the files you have up, and maybe I'm not fully understanding, but will I be able to set up this server with the OG files if the OG files are not available for download?

On the copyright front, are you already in contact with the current copyright holders? If you think messages from the community might help move that process along, I'm happy to get some testimonials.

Cheers,
Mike

**Reply by Fruktus** [reply]
Hi. The files here are only for server, since I can't upload the OG here myself, but at this point I think I'll just link them on Reddit, since you can't really get all of them anywhere now and I've made quite an extensive backup of basically everything I could find before it went down :) 

On the copyright front I've tried mailing and someone on Reddit said that they tried contacting in person but I received no response. Maybe I got the wrong mail or maybe the author is not interested, I do not know. I do know however that from all the places that QR was active, the Facebook page seems to be still up and there was some activity during Covid pandemic, so maybe I'll try my luck there.

Best regards,
Fruktus

**Reply by Fruktus** [reply]
Also, if you feel like asking yourself absolutely try, maybe more voices will help :)

**Reply by mstahly** [reply]
![Quadradius Convo](https://github.com/Fruktus/QuadradiusPreservationProject/assets/150499565/ba42ecd6-744a-478c-a993-3b3da4316dd3)

You can contact the Quadradius guys here: jimmi.heiserman@gmail.com  or through their FB (see screenshot that I think attached correctly)

**Reply by Fruktus** [reply]
Holly molly, you have done it. I've sent the email already, I'll post an update if I get a response. It seems that indeed I had a wrong email adress. Thank you so much for help.

**Reply by mstahly** [reply]
Yes!! This makes me happy

**Reply by Fruktus** [reply]
FYI I got a response to my initial email, I'm waiting for followup, will post when I get update

**Reply by mstahly** [reply]
Right on - thanks for the update. I'm excited to see what happens

**Comment by ayush194** [comment]
This is sooo cool guys!! Have been waiting for this for years! Let me know
if I can help in any way. I can code pretty well.

Ayush

**Reply by Fruktus** [reply]
Sure will if we'll get the blessing :D

**Comment by Fruktus** [comment]
Hi, a followup - I have received permission to continue with the project. 

Quote from conversation:
`If you have any files saved or files that you started to implement, you are free to experiment with that, put them online , anything`.

So, in next couple of days I'll try to prepare additional repo just for those original files, I'll then link it to this one, so the full app will be buildable (the OG classic) and I'll update the repo (readme is outdated and I found some bugs on the way).

Up until 21st I'm going to be somewhat busy, but I'll try to at least have everything online.

Thanks everyone for help, you're awesome :muscle:

**Reply by mstahly** [reply]
Let's goooooo!

**Comment by ayush194** [comment]
Awesome!! Let’s host this great game again!

**Comment by Fruktus** [comment]
Old files are available here, I've added a disclaimer that this is uploaded by me (third party) and not the author, so hopefully no one will bother him needlessly.

My contributing friend @kjarosh is working on updating the docker image and hopefully it will get uploaded soon.
[Classic Quadradius](https://github.com/Fruktus/QuadradiusClassic/tree/master)

**Reply by Fruktus** [reply]
On a sidenote - hosting this is somewhat problematic, as it requires a server (which costs money), and even then the players would need to obtain Flash Projector, which to my best knowledge had been taken down from official sites, and I'm not sure if third parties can be trusted. I have my own copies and if needed I can provide their checksums for verification if anyone needs them.

**Comment by ayush194** [comment]
Hi,

Just one question. Do we have a plan to set up a free central web server
and host this game so everyone online can play each other?
To do this free of cost, we could probably host a free web server on Heroku.

Thanks!
Ayush

**Reply by Fruktus** [reply]
Ah, it seems we thought of the same thing - https://github.com/Fruktus/QuadradiusPreservationProject/discussions/26#discussioncomment-7781022

**Reply by Fruktus** [reply]
So I don't really know how to reasonably approach it for the reasons above, mainly the flash. That's basically why we started migrating it. Unless the initiatives like Ruffle.js were updated and would work (when I tested them about year or two ago it did not work)

**Reply by ayush194** [reply]
What about trying to find something free. Here is a list of free Python server hosting https://wiki.python.org/moin/FreeHosts.
Open to other suggestions. Heroku also offers free Apache PHP servers, not sure about Python servers.

**Reply by Fruktus** [reply]
Heroku perhaps could be done, although I think they may not be allowing port forwarding, which would be needed if we wanted to use original files (unmodded). I don't know if it would be possible to change them not to require this.

**Reply by Fruktus** [reply]
Straightforward option would be to use something like AWS

**Reply by ayush194** [reply]
Yes, but AWS will never be free actually.

**Reply by ayush194** [reply]
What about other options like these: https://wiki.python.org/moin/FreeHosts?

**Reply by Fruktus** [reply]
Yea well, depending on the cost maybe I could manage. I'll check that out later. Anyway from what I see Ruffle was updated so I'll need to check if something has changed and maybe it can work now. That would make things so much easier.

**Reply by ayush194** [reply]
If I have your permission to use your repository, I could also explore these and will update this thread if I can set something up free of cost.

**Reply by Fruktus** [reply]
Absolutely go for it, this one has license added which basically lets you do whatever you want with the code here. The OG Files do not have license since its not my code and I don't know which is applicable, but I'm pretty sure its ok for something like that

**Reply by Fruktus** [reply]
I've tested some of the free hosts, but sadly they either have very restrictive limits, like 100s of cpu time, which is roughly a couple of minutes or do not allow port forwarding. I'll look for alternatives

**Comment by Fruktus** [comment]
OMG it looks like ruffle is indeed working now. So basically you don't need any 3rd party apps, you can just straight up play in browser. Time for updates <3

**Comment by Fruktus** [comment]
@mstahly @ayush194 We've set up vps for testing, you can try it out at ```quadradius.ddns.net```, ports 3000 and 3001 (default), easiest way is to just enter this address into the launcher, which you can download from [releases](https://github.com/Fruktus/QuadradiusPreservationProject/releases/tag/v1.0.0) page. You can test the game if you'd like, if you will and notice any bugs let me know. Since this is the first time I'm setting up something like this expect things to break and some downtimes. I'll try to update it and maintain it for now. You can play as members - use any username and password you'd like (keep in mind that they are not encrypted so you can pass any gibberish you'd like), the accounts are not saved and neither are scores. If you have any questions le me know.

**Reply by mstahly** [reply]
Amazing! Will test as soon as I can get someone to play with me

**Reply by Fruktus** [reply]
Thanks! Hope you have fun

**Reply by mstahly** [reply]
I was able to join the lobby earlier today and try to initiate a game with someone who happened to be in there. Spun but did not load the game itself. Now when I go to the lobby, it doesn't register me as being there in the player list. Sharing these details in case they're helpful

**Reply by Fruktus** [reply]
Thanks, I'll look into it! Although that is interesting. Were you using launcher or something else?

**Reply by Fruktus** [reply]
Nvm, I see the error, will fix soon

**Reply by Fruktus** [reply]
Most likely cause is the usage of non-ascii characters in nickname or the chat message. Flash restricted this but ruffle currently does not. Any idea what might've been the problem? For now if you'd like to play you can try avoiding special characters, we'll try to work around this problem until ruffle adds proper support for restrictions.

**Reply by mstahly** [reply]
aha - gotcha. thanks!

**Reply by Fruktus** [reply]
Hi! Thanks to the work of @kjarosh, the web version is now functional - you can just open the address in the browser and play straight from it, no need for launcher.

**Reply by mstahly** [reply]
Are you serious?! This is amazing. You all are amazing :)
